app/src/main/java/com/android/sample/ui/listActivities/ViewListActivities.kt

```diff
@@ -183,18 +207,44 @@ fun ActivityCard(
           }
 
           Spacer(modifier = Modifier.height(8.dp))
+          Row(
+              modifier = Modifier.padding(horizontal = 16.dp).fillMaxWidth(),
+              horizontalArrangement = Arrangement.SpaceBetween,
+              verticalAlignment = Alignment.CenterVertically) {
+                // Display the date
+                Text(
+                    text = formattedDate,
+                    style =
+                        MaterialTheme.typography.bodySmall.copy(
+                            color = Color.Gray, // Light gray color for the date
+                            fontStyle = FontStyle.Italic),
+                    modifier = Modifier.weight(1f) // Takes up remaining space
+                    )
 
-          // Display the date
-          Text(
-              text = formattedDate,
-              style =
-                  MaterialTheme.typography.bodySmall.copy(
-                      color = Color.Gray, // Light gray color for the date
-                      fontStyle = FontStyle.Italic),
-              modifier = Modifier.padding(horizontal = 16.dp))
+                if (profile != null) {
+                  IconButton(
+                      onClick = {
+                        isLiked = !isLiked
+                        if (isLiked) {
+                          profileViewModel.addLikedActivity(profile.id, activity.uid)
+                        } else {
+                          profileViewModel.removeLikedActivity(profile.id, activity.uid)
+                        }
+                      },
+                      modifier = Modifier.testTag("likeButton$isLiked"),
+                  ) {
+                    Icon(
+                        imageVector =
+                            if (isLiked) Icons.Filled.Favorite else Icons.Outlined.FavoriteBorder,
+                        contentDescription = if (isLiked) "Liked" else "Not Liked",
+                        tint = if (isLiked) Color.Black else Color.Gray,
+                    )
+                  }
+                }
+              }
```

Comment:
The button toggles the isLiked state and updates the database using profileViewModel based on whether an activity is liked or unliked, which is straightforward and effective.